File: map2d/src/test/java/net/ponderingprogrammer/map2d/ConcreteFixedSquareMap2d.java
package net.ponderingprogrammer.map2d;

import java.util.Collection;

/**
 * For testing FixedSquareMap2d
 */
public class ConcreteFixedSquareMap2d extends FixedSquareMap2d<Integer> {
    public ConcreteFixedSquareMap2d(int width, int height) {
        super(width, height);
    }

    @Override
    public Collection<Cell<Integer>> findAdjacentCells(int x, int y) {
        return null;
    }

    @Override
    public CellGraph<Integer> toCellGraph() {
        return null;
    }

}

File: NProcGen/tests/PonderingProgrammer.Map2dTests/ConcreteFixedSquareMap2d.cs
using PonderingProgrammer.Map2d;
using System.Collections.Generic;

namespace PonderingProgrammer.Map2dTests
{
    class ConcreteFixedSquareMap2d : FixedSquareMap2d<bool>
    {
        public ConcreteFixedSquareMap2d(int width, int height) : base(width, height)
        {
        }

        public override IEnumerable<Cell<bool>> FindAdjacentCells(int x, int y)
        {
            return null;
        }

        public override CellGraph<bool> ToCellGraph()
        {
            return null;
        }
    }
}
